src/Tuura/Concept/STG/Translation.hs

```diff
@@ -51,7 +47,7 @@ validate signs circuit
     inconsistent = filter ((==Inconsistent) . initial circuit) signs
     undef        = filter ((==Undefined) . initial circuit) signs
 
-handleArcs :: [([Transition Signal], Transition Signal)] -> [String]
+handleArcs :: Show a => [([a], a)] -> [String]
```

Comment:
I suggest to use `Transition a` here instead of just `a`. We don't have to make the code as abstract as possible, but abstract enough so we don't have to duplicate things. I think `Transition` does't get in the way of reusing code. In fact, it acts as documentation (clarifies the meaning of the code), plus allows you to use things like `newValue`.